runtime/compiler/optimizer/J9ValuePropagation.cpp

```diff
@@ -3782,9 +3782,23 @@ bool J9::ValuePropagation::isUnreliableSignatureType(
       return false;
 
    int32_t numDims = 0;
+   TR_OpaqueClassBlock *originClass = klass;
    klass = comp()->fej9()->getBaseComponentClass(klass, numDims);
    if (!TR::Compiler->cls.isInterfaceClass(comp(), klass))
+      {
+      // If the original class is an array class and it is not a null-restricted array,
+      // we can not trust its type because it can be a nullable array class type or a null-restricted array type
+      if (TR::Compiler->om.areFlattenableValueTypesEnabled() &&
+         (numDims > 0) &&
+         TR::Compiler->cls.isValueTypeClass(klass) &&
+         !TR::Compiler->cls.isArrayNullRestricted(comp(), originClass))
+         {
+         erased = NULL;
+         return true;
+         }
```

Comment:
I think I understand what you mean here. Instead of erasing all information about null-restricted arrays, we can treat them like interface arrays to be handled in the later of the method. Just to double check, is it something like below? 

```c++
   if (!TR::Compiler->cls.isInterfaceClass(comp(), klass))
      {
      // If the original class is an array class and it is not a null-restricted array,
      // we can not trust its type as a nullable array because it can either be a nullable
      // array class type or a null-restricted array type.
      if (TR::Compiler->om.areFlattenableValueTypesEnabled() &&
         (numDims > 0) &&
         TR::Compiler->cls.isValueTypeClass(klass) &&
         !TR::Compiler->cls.isArrayNullRestricted(comp(), originClass))
         {
         // Do nothing here and will be handled next like an interface array
         }
      else
         {
         return false;
         }
      }
```

Reply:
Yes, that's what I was thinking.

Reply:
Addressed in [49082a](https://github.com/eclipse-openj9/openj9/compare/79e5a3e7017980af6bb9b4cfc3d473a880162938..49082abb276d98df16ae48d371468818d32bcf7f). Ready for another review. Thank you!